src/modules/poll/index.js

```diff
@@ -0,0 +1,303 @@
+        return response;
+    }
+
+    if (!poll.isOpen) {
+        response.output = i18n('poll.alreadyclosed');
+        return response;
+    }
+
+    const newList = pollsList.filter(p => p.pollId !== requestedPollId);
+
+    poll.isOpen = false;
+
+    newList.push(poll);
+
+    if (!isEqual(pollsList, newList)) {
+        updateModuleData('poll', {
+            pollsList: newList,
+        });
+    }
+
+    response.output = i18n('poll.close', { requestedPollId });
+    return response;
+};
+
+const vote = async function (response, {
+    i18n,
+    id,
+    getModuleData,
+    updateModuleData,
+}) {
+    const { votesList = [], pollsList = [] } = await getModuleData('poll');
+    const { args: { requestedPollId, requestedOption } } = response;
+
+    const poll = pollsList.find(p => p.pollId === requestedPollId);
+
+    if (!poll) {
+        response.output = i18n('poll.notfound', { requestedPollId });
+        return response;
+    }
+
+    if (!poll.isOpen) {
+        response.output = i18n('poll.alreadyclosed');
+        return response;
+    }
+
+    if (votesList.find(v => (v.pollId === requestedPollId && v.voterId === id))) {
+        response.output = i18n('poll.alreadyvoted');
+        return response;
+    }
+
+    if (poll.options.includes(requestedOption)) {
+        const newVote = {
+            voterId: id,
+            pollId: requestedPollId,
+            option: poll.options.indexOf(requestedOption),
+            dateVoted: new Date(),
+        };
+        updateModuleData('poll', {
+            votesList: [...votesList, newVote],
+        });
+        const optionText = requestedOption;
+        response.output = i18n('vote.cast', {
+            id,
+            requestedPollId,
+            optionText,
+        });
+        return response;
+    }
+
+    if (requestedOption - 1 >= 0 && requestedOption - 1 < poll.options.length) {
+        const newVote = {
+            voterId: id,
+            pollId: requestedPollId,
+            option: parseInt(requestedOption, 10),
+            dateVoted: new Date(),
+        };
+        updateModuleData('poll', {
+            votesList: [...votesList, newVote],
+        });
+        const optionText = poll.options[requestedOption - 1];
+        response.output = i18n('vote.cast', {
+            id,
+            requestedPollId,
+            optionText,
+        });
+        return response;
+    }
+
+    response.output = i18n('vote.nosuchoption');
+    return response;
+};
+
+const checkVote = async function (response, {
+    getModuleData,
+    updateModuleData,
+    input,
+    id,
+    i18n,
+}) {
+    const { pollsList = [], votesList = [] } = await getModuleData('poll');
+
+    if (!pollsList.find(poll => poll.isOpen)) {
+        return response;
+    }
+
+    pollsList.filter(p => p.isOpen).forEach((p) => {
+        if (votesList.find(v => (v.pollId === p.pollId && v.voterId === id))) {
+            return response;
```

Comment:
return response from () => {} ??? what u want? 

need simple return to prevent below code exec? or u want prevent forEach?

Reply:
I want to skip to the next iteration if the condition is true
same in the following if's inside this function